stl/inc/expected

```diff
@@ -685,6 +689,303 @@ public:
         }
     }
 
+    template <class _Uty = _Err>
+    _NODISCARD constexpr _Err error_or(_Uty&& _Other) const& noexcept(
+        is_nothrow_copy_constructible_v<_Err>&& is_nothrow_convertible_v<_Uty, _Err>) { // strengthened
+        static_assert(
+            is_copy_constructible_v<_Err>, "is_copy_constructible_v<E> must be true. (N4928 [expected.object.obs]/20)");
+        static_assert(
+            is_convertible_v<_Uty, _Err>, "is_convertible_v<U, E> must be true. (N4928 [expected.object.obs]/20)");
+
+        if (_Has_value) {
+            return _STD forward<_Uty>(_Other);
+        } else {
+            return _Unexpected;
+        }
+    }
+
+    template <class _Uty = _Err>
+    _NODISCARD constexpr _Err error_or(_Uty&& _Other) && noexcept(
+        is_nothrow_move_constructible_v<_Err>&& is_nothrow_convertible_v<_Uty, _Err>) { // strengthened
+        static_assert(
+            is_move_constructible_v<_Err>, "is_move_constructible_v<E> must be true. (N4928 [expected.object.obs]/22)");
+        static_assert(
+            is_convertible_v<_Uty, _Err>, "is_convertible_v<U, E> must be true. (N4928 [expected.object.obs]/22)");
+
+        if (_Has_value) {
+            return _STD forward<_Uty>(_Other);
+        } else {
+            return _STD move(_Unexpected);
+        }
+    }
+
+    // [expected.object.monadic]
+
+    template <class _Fn>
+        requires is_copy_constructible_v<_Err>

```

Comment:
We guard concepts usage with `#ifdef __cpp_lib_concepts`, even in C++23.
This is needed for EDG target.  See similar pieces of `<optional>`.
Alternatively, consider avoiding concepts if it is doable with a reasonable small effort.

Reply:
The whole file is already guarded with `__cpp_lib_concepts`.
https://github.com/microsoft/STL/blob/992efd9f9f96d1ed6f041332a4b1be1f31b8de02/stl/inc/expected#L11-L14